server/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -1072,6 +1077,28 @@ private void fillSegmentInfo(SegmentReader segmentReader, boolean search, Map<St
         }
         segment.segmentSort = info.info.getIndexSort();
         segment.attributes = info.info.getAttributes();
+        Codec codec = info.info.getCodec();
+        segment.codec = codec.getName();
+        if (codec instanceof PerFieldMapperCodec) {
+            try {
+                // potentially expensive with many fields
+                FieldInfos fieldInfos = segmentReader.getFieldInfos();
+                if (fieldInfos.hasVectors()) {
+                    for (FieldInfo fieldInfo : fieldInfos) {
```

Comment:
while I think this shouldn't be problematic in many cases, it still might be a bit expensive in case there are many vector fields in many segments.